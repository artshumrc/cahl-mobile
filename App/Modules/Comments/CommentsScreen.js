import React from 'react';
import { View, ScrollView, TextInput, Image, Button, Text } from 'react-native';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';
import firebase from 'firebase';

// components
import Comment from './Comment';


// styles
import styles from './CommentsScreenStyles';

class CommentsScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      content: 'Add your comment.',
      currentUser: firebase.auth().currentUser,
    };

    this.post = this.post.bind(this);
    this.login = this.login.bind(this);
  }

  static navigationOptions = {
    title: 'Comments'
  };

  login() {
    const { navigate } = this.props.navigation;
    navigate('LoginScreen');
  }

  post() {
    const { mutate, navigation, data } = this.props;
    const { currentUser, content } = this.state;

    alert('Comments are made public upon posting!');

    mutate({
      variables: {
        itemId: navigation.state.params.recordId,
        userDisplayName: currentUser.displayName,
        content: content,
        photoURL: currentUser.photoURL,
      }
    }).then(({ data }) => console.log('success! ', data))
      .catch(error => console.log('error: ', error));
    data.refetch();
  }

  render() {
    const { content, currentUser } = this.state;
    const { data } = this.props;

    return (
      <View style={styles.container}>
        <ScrollView>
          {
            data.comments && data.comments.length > 0 ?
              <View>
                {
                  data.comments.map(comment =>
                    <Comment
                      key={comment._id}
                      content={comment.content}
                      displayName={comment.userDisplayName}
                      createdAt={comment.createdAt}
                      photoURL={comment.photoURL}
                    />
                  )
                }
              </View>
              :
              <View>
                <Text>No comments available.</Text>
              </View>
          }
        </ScrollView>
        <View style={styles.textInputContainer}>
          {
            currentUser &&
              <Image
                style={styles.image}
                source={{ uri: currentUser.photoURL }}
              />
          }
          <TextInput
            value={content}
            onChangeText={(content) => this.setState({ content })}
            style={styles.textInput}
            clearTextOnFocus={true}
          />
          {
            currentUser ?
              <Button
                title={'Post'}
                onPress={this.post}
              />
            :
              <Button
                title={'Sign in'}
                onPress={this.login}
              />
          }
        </View>
      </View>

    );
  }
}

const getComments = gql`
query getComments($itemId: ID!) {
  comments(itemId: $itemId) {
    content,
    createdAt,
    userDisplayName,
    photoURL,
  }
}
`;

const addComment = gql`
mutation content($itemId: String!, $userDisplayName: String!, $content: String!, $photoURL: String!) {
  commentCreate(itemId: $itemId, userDisplayName: $userDisplayName, content: $content, photoURL: $photoURL) {
    content,
    itemId,
    userDisplayName,
    createdAt,
    photoURL,
  }
}
`;

export default compose(
  graphql(getComments, {
    options: ownProps => ({
      variables: {
        itemId: ownProps.navigation.state.params.recordId,
      }
    }),
  }),
  graphql(addComment),
)(CommentsScreen);
