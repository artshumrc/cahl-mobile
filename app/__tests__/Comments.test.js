import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CommentsScreen from '../modules/Comments';

it('renders correctly', () => {
  const tree = renderer.create(
    <CommentsScreen />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
