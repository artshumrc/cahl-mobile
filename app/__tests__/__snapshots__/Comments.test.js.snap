// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/IMG_0487.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 25,
                "height": 50,
                "margin": 10,
                "width": 50,
              }
            }
          />
          <View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "marginTop": 10,
                  "width": 300,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontFamily": "AvenirNext-bold",
                    }
                  }
                >
                  Tyler Kirby  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontFamily": "AvenirNext-regular",
                    }
                  }
                >
                  Consectetur culpa amet sint non tempor labore commodo ex in labore in minim officia laborum. Ullamco nulla magna duis dolore incididunt incididunt aute eu aliquip ex aliqua amet magna.
                </Text>
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                }
              }
            >
              July 26, 2017
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/IMG_0487.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 25,
                "height": 50,
                "margin": 10,
                "width": 50,
              }
            }
          />
          <View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "marginTop": 10,
                  "width": 300,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontFamily": "AvenirNext-bold",
                    }
                  }
                >
                  Tyler Kirby  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontFamily": "AvenirNext-regular",
                    }
                  }
                >
                  Minim minim ea consectetur eu deserunt cillum. Consectetur incididunt ad adipisicing nulla aute consequat. Magna reprehenderit ex veniam do esse tempor. Ipsum proident ipsum laborum deserunt nulla enim esse. Proident excepteur tempor anim laborum do aute elit non tempor mollit sunt mollit. Officia irure in qui occaecat est anim commodo anim ut eiusmod adipisicing eu ad est.
                </Text>
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                }
              }
            >
              July 29, 2017
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/IMG_0487.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "marginRight": 10,
          "width": 30,
        }
      }
    />
    <TextInput
      clearTextOnFocus={true}
      onChange={[Function]}
      style={
        Object {
          "color": "#999999",
          "height": 40,
          "width": 250,
        }
      }
      value="Add your comment."
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign in
        </Text>
      </View>
    </View>
  </View>
</View>
`;
